model/DataObject.php

```diff
@@ -3433,7 +3435,10 @@ public function databaseIndexes() {
 					if ($table && strtolower($table) !== strtolower($this->class)) {
 						continue;
 					}
-
+					// Skip primary key; Already indexed
+					if ($column === 'ID') {
```

Comment:
I don't understand why "already indexed" is an exclusion criteria here. Wouldn't you expect `ID` to be part of the returned set?

Reply:
Because you shouldn't add a second index when the ID is already the primary key.

Reply:
Let me look into it more.

Reply:
Initial investigation is that this is a pain to fix for all fixed_fields unless I add heaps more code, and we don't really need to return ID as an indexed field (it's hard-coded as an index in 3.x anyway).

Reply:
ok did some more work. Should be ok now.